src/mlpack/tests/ksinit_test.cpp

```diff
@@ -0,0 +1,315 @@
+#include <mlpack/core/optimizers/rmsprop/rmsprop.hpp>
+#include <mlpack/methods/ann/layer/layer.hpp>
+
+#include <boost/test/unit_test.hpp>
+#include "test_tools.hpp"
+
+using namespace mlpack;
+using namespace mlpack::ann;
+using namespace mlpack::optimization;
+
+BOOST_AUTO_TEST_SUITE(KSInitialization);
+
+/**
+ * Train and evaluate a vanilla network with the specified initialisation 
+   procedure.
+ */
+template<typename MatType = arma::mat>
+void BuildVanillaNetwork(MatType& trainData,
+                         MatType& trainLabels,
+                         MatType& testData,
+                         MatType& testLabels,
+                         const size_t hiddenLayerSize,
+                         const size_t maxEpochs,
+                         const size_t outputSize,
+                         double& trainError,
+                         double& testError)
+{
+  /*
+  @param trainError mean squared error of predictions on training data.
+  @param testError  mean squared error of predictions on test data.
+   * Construct a feed forward network with trainData.n_rows input nodes,
+   * hiddenLayerSize hidden nodes and trainLabels.n_rows output nodes. The
+   * network structure looks like:
+   *
+   *  Input         Hidden        Output
+   *  Layer         Layer         Layer
+   * +-----+       +-----+       +-----+
+   * |     |       |     |       |     |
+   * |     +------>|     +------>|     |
+   * |     |     +>|     |     +>|     |
+   * +-----+     | +--+--+     | +-----+
+   *             |             |
+   *  Bias       |  Bias       |
+   *  Layer      |  Layer      |
+   * +-----+     | +-----+     |
+   * |     |     | |     |     |
+   * |     +-----+ |     +-----+
+   * |     |       |     |
+   * +-----+       +-----+
+   */
+
+  //4.59 is a constant used in the paper.
+  KathirvalavakumarSubavathiInitialization init(trainData, 4.59);
+
+  FFN<MeanSquaredError<>, KathirvalavakumarSubavathiInitialization> 
+      model(MeanSquaredError<>(), init);
+  
+  model.Add<Linear<> >(trainData.n_rows, hiddenLayerSize);
+  model.Add<LeakyReLU<> >();
+  model.Add<Linear<> >(hiddenLayerSize, outputSize);
+
+  RMSprop<decltype(model)> opt(model, 0.01, 0.88, 1e-8,
+      maxEpochs * trainData.n_cols, 1e-18);
+
+  model.Train(std::move(trainData), std::move(trainLabels), opt);
+
+  MatType prediction;
+ 
+  // Calculating the mean squared error on the training data.
+  model.Predict(trainData, prediction);
+
+  trainError = arma::mean(arma::mean(arma::square(prediction - trainLabels)));
+  // Calculating the mean squared error on the test data
+  model.Predict(testData, prediction);
+  testError = arma::mean(arma::mean(arma::square(prediction - testLabels)));
+  
+}
+
+/* Error is a structure which has the MSE of Training data and MSE of 
+   Validation data respectively. This structure is returned to the parent test 
+   case by CrossValidation function.
+
+*/
+struct Error
+{
+  double trainError;
+  double validationError;
+};
```

Comment:
I guess, just to be consistent with the rest of the code style used in your test, I would go with a reference for ``trainError`` and ``validationError`` as you have done for the BuildVanillaNetwork function. What do you think?